helix-term/src/args.rs

```diff
@@ -88,7 +88,12 @@ impl Args {
 
 /// Parse arg into [`PathBuf`] and position.
 pub(crate) fn parse_file(s: &str) -> (PathBuf, Position) {
-    let def = || (PathBuf::from(s), Position::default());
+    let def = || {
+        (
+            helix_core::path::expand_tilde(Path::new(s)),
```

Comment:
I think it would be better to call this from `open` in `helix-term/src/commands/typed.rs`. We don't need to do this for files passed in the argv since the shell will do the tilde expansion and there's a small bug here where `:open` fails if you give it a directory that has a tilde and a row or row/col position (though that isn't a use-case that makes sense)

Reply:
Okay, makes sense 👍